### ISSUE TITLE
EasyBuild/3.5.3, Bison-3.0.4-GCCcore-4.9.3.eb, sanity check failed - lib vs. lib64

### ISSUE THREAD

**Issue by pmatousu** [easybuilders/easybuild-easyconfigs#6165]
I am trying to build some gcc with higher version than 4.8, in my latest try with     ISL-0.16-GCC-4.9.3-2.25.eb 

    9971 >eb Bison-3.0.4-GCCcore-4.9.3.eb -r
    == temporary log file in case of crash /tmp/eb-GPpgby/easybuild-FTgRr8.log
    == resolving dependencies ...
    == processing EasyBuild easyconfig /home/petrm/opt/eb/software/EasyBuild/3.5.3/lib/python2.7/site-packages/easybuild_easyconfigs-3.5.3-py2.7.egg/easybuild/easyconfigs/b/Bison/Bison-3.0.4-GCCcore-4.9.3.eb
    == building and installing Bison/3.0.4-GCCcore-4.9.3...
    == fetching files...
    == creating build dir, resetting environment...
    == unpacking...
    == patching...
    == preparing...
    == configuring...
    == building...
    == testing...
    == installing...
    == taking care of extensions...
    == postprocessing...
    == sanity checking...
    == FAILED: Installation ended unsuccessfully (build directory: /home/petrm/Documents/build/eb/build/Bison/3.0.4/GCCcore-4.9.3): build failed (first 300 chars): Sanity check failed: no file of ('lib/liby.a',) in /home/petrm/opt/eb/software/Bison/3.0.4-GCCcore-4.9.3
    == Results of the build can be found in the log file(s) /tmp/eb-GPpgby/easybuild-Bison-3.0.4-20180417.132532.OKzbi.log
    ERROR: Build of /home/petrm/opt/eb/software/EasyBuild/3.5.3/lib/python2.7/site-packages/easybuild_easyconfigs-3.5.3-py2.7.egg/easybuild/easyconfigs/b/Bison/Bison-3.0.4-GCCcore-4.9.3.eb failed (err: "build failed (first 300 chars): Sanity check failed: no file of ('lib/liby.a',) in /home/petrm/opt/eb/software/Bison/3.0.4-GCCcore-4.9.3")

But:

    9980 >file /home/petrm/opt/eb/software/Bison/3.0.4-GCCcore-4.9.3/lib64/liby.a
    /home/petrm/opt/eb/software/Bison/3.0.4-GCCcore-4.9.3/lib64/liby.a: current ar archive

So lib64 instead of lib.
I see that often on my system with easybuild.

**Comment by pmatousu** [comment]
Very same issue in GMP-6.1.0-GCC-4.9.3-2.25.eb
lib instead lib64
Even here I can correct it easily myself.

**Comment by boegel** [comment]
@pmatousu We indeed have `lib/` hardcoded in too many places, which causes problems on SUSE systems.

For this particular case, you can fix it by changing the Bison easyconfig file as follows:

```diff
diff --git a/easybuild/easyconfigs/b/Bison/Bison-3.0.4-GCC-4.9.3.eb b/easybuild/easyconfigs/b/Bison/Bison-3.0.4-GCC-4.9.3.eb
index 55b6f06ba..dfc3ed912 100644
--- a/easybuild/easyconfigs/b/Bison/Bison-3.0.4-GCC-4.9.3.eb
+++ b/easybuild/easyconfigs/b/Bison/Bison-3.0.4-GCC-4.9.3.eb
@@ -15,7 +15,7 @@ source_urls = [GNU_SOURCE]
 builddependencies = [('M4', '1.4.17')]
 
 sanity_check_paths = {
-    'files': ["bin/%s" % x for x in ["bison", "yacc"]] + ["lib/liby.a"],
+    'files': ["bin/%s" % x for x in ["bison", "yacc"]] + [('lib/liby.a', 'lib64/liby.a')],
     'dirs': [],
 }
 ```

That makes EasyBuild consider both `lib/liby.a` and `lib64/liby.a`, and the check will pass if either of those is available.

Doing this on a case-by-case basis is going to be painful though, so there have been proposals in the past to deal with this centrally instead, see for example the promising proposed change in https://github.com/easybuilders/easybuild-framework/issues/1404/files.
That never got merged however,

**Comment by pmatousu** [comment]
Actually I did it by very similar way and it works ok. Thank you for including diff.

**Comment by boegel** [comment]
fixed in #6170
